src/main/scala/mesosphere/marathon/MarathonConf.scala

```diff
@@ -20,6 +20,10 @@ import org.rogach.scallop.ScallopConf
 
 import scala.sys.SystemProperties
 
+object FancyHostNameCache {
```

Comment:
The `hostname` will be advertised in the cluster, so "localhost" is not a good idea unless everything is local. 
Please:
- move this property into the definition of the lazy val, so it can not be reached anywhere else
- make it lazy

Reason:
1) Nobody should use something like `java.net.InetAddress.getLocalHost.getHostName` directly - everybody should use the hostname property.
2)There are systems that do not have resolvable host names. This method will throw an exception in this case.